### PR TITLE
fix: move data fetching into card line chart to prevent rendering loop on contributor profile

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Fixes a rendering issue with the pull requests chart for a contributor, causing navigation after visiting a user profile to stop working until a page refresh

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes https://github.com/open-sauced/app/issues/2656
Fixes https://github.com/open-sauced/app/issues/2622

## Mobile & Desktop Screenshots/Recordings

BEFORE:

https://github.com/open-sauced/app/assets/42211/69fe9d36-75db-46df-82ee-61aca8310f1e

AFTER:

https://github.com/open-sauced/app/assets/42211/93c94ddf-cee4-42b7-bce1-8245832832ce

## Steps to QA

1. Navigate to the deploy preview
2. Visit a contributor profile - such as /user/brandonroberts
3. Click through the profile tabs for Highlights, Contributions
4. Navigate away from the page

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/organisms/ContributorProfileTab/contributor-profile-tab.tsx [hunk 1]
```diff
@@ -403,7 +401,7 @@ const ContributorProfileTab = ({
               </div>
             </div>
             <div className="mt-2 h-36">
-              <CardLineChart lineChartOption={chart} className="!h-36" />
+              <CardLineChart contributor={githubName} repoIds={[]} className="!h-36" />
```
Comment (nickytonline) [inline components/organisms/ContributorProfileTab/contributor-profile-tab.tsx:404]: refactor (nit): Any reason the `repoIds` is always an empty array? If the field is optional, consider adding a default value if the prop isn't required.

**Review by nickytonline — COMMENTED** [review]
This looks great

**Comment by nickytonline** [comment]
So is the fix we no longer pass in a chart to the `<CardLineChart />` component? Just trying to understand what was causing the rendering issue.

**Comment by brandonroberts** [comment]
> So is the fix we no longer pass in a chart to the `<CardLineChart />` component? Just trying to understand what was causing the rendering issue.

I think the root cause was a combination of that and the `useEffect` for the chart data hook causing it to rerender endlessly.

**Review by nickytonline — APPROVED** [review]
:shipit:

**Pull request merged**